### PR TITLE
App Launcher

### PR DESCRIPTION
**Pull request by dlabrecq**
Added app launcher from ngx-forge.

Note that this is a work in progress. The _applauncher route is defined as an experimental feature; however, the feature's user-enabled flag is still false -- need some help setting that up

### CODE REVIEW

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2466-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-ui.badger.fabric8.io

**Comment by gastaldi** [comment]
Make sure to align the services json with the models available in https://app.swaggerhub.com/apis/fabric8-launcher/backend/2.0.0

**Comment by bartoszmajsak** [comment]
Why is front-end [calling GH directly](https://github.com/fabric8-ui/fabric8-ui/pull/2466/files#diff-443e9a393085d9286e0795e3f8b34316R18)? Is that a temporary solution or do we want to use it this way? @gastaldi isn't it responsibility of the launcher backend?

File: src/app/app-launcher/services/app-launcher-gitprovider.service.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  GitHubRepoDetails,
+  GitHubRepoLastCommit,
+  GitHubRepoLicense,
+  GitHubUser
+} from '../../space/create/codebases/services/github';
+
+// Enable Access-Conrtol-Expose-Headers for CORS to test
+@Injectable()
+export class AppLauncherGitproviderService implements GitProviderService {
+  private static readonly HEADERS = new Headers({
```
Comment (edewit) [inline src/app/app-launcher/services/app-launcher-gitprovider.service.ts]: The backend already does something like this, I don't think we need this functionality in the front-end

**Comment by corinnekrych** [comment]
@dlabrecq see https://github.com/dlabrecq/fabric8-ui/pull/1 after that you should see your new menu, AppLauncher feature is corrlty set in PROD and PRE-PROD

File: src/app/app-launcher/services/app-launcher-gitprovider.service.ts (hunk 1 above)
Comment (dlabrecq) [reply src/app/app-launcher/services/app-launcher-gitprovider.service.ts]: No, this was just taken from the demo. Ultimately, the providers will be written for OSIO.

The main reason this PR was created is so Corinne can help make this an experimental feature. Updates to the providers will follow.

**Comment by dlabrecq** [comment]
@corinnekrych I don't believe we want to have a menu item to navigate to the app launcher.

The idea is to show the app launcher after adding a space, similar to how the forge wizard is shown after a space is created. When the new dialog/overlay is shown, the next step in the workflow will be to show the full page app launcher.

Is there a way to make this an experimental feature, but not require the user to navigate to a space?

**Comment by corinnekrych** [comment]
@dlabrecq 
ah sorry close my PR then. The routing as you did should be good.

**Comment by dlabrecq** [comment]
Update: I'm able to "opt-in to internal features" now and navigate to the App Launcher as expected.

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2466-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-ui.badger.fabric8.io

**Comment by dlabrecq** [comment]
Updated ngx-forge to v0.0.39-development

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2466-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-ui.badger.fabric8.io

**Pull request merged**